Dockerfile

```diff
@@ -1,22 +1,33 @@
-# golang1.8.1 base image
-FROM golang:1.8.1
+# Build backend
+FROM golang:1.8.1 as backend-build
 
-# copy source
 COPY . /go/src/commento
 WORKDIR /go/src/commento
 
-# build backend
 RUN go get -v .
-RUN go install .
+RUN go build -ldflags '-linkmode external -extldflags -static -w'
```

Comment:
Not a review comment, just curious: what does this do? I haven't been writing Go long enough ^^

Searching tells me `-linkmode external` can be used to load go variables at compile time. Is that the reasoning?

`-static` is static linking, I can tell. `-w` is some kind of executable size reduction technique from what I read (add `-s` as well?).

Reply:
The reason I went with this versus a standard `go build` is because I wanted to build a static binary from the `golang` Docker image, but still be able to use it in `alpine`. There are a couple of problems if we rely on the former.

One of which is your project's use of `mattn/go-sqlite3`. It is a `cgo` package, not a Go package. Your project however, is in Go. So we have situation where we need a combination of Go, and non-Go code to be in the final executable without dynamic libraries. So, `-linkmode external` simply tells `cmd/linker` to invoke the host linker (often `gcc`) to combine your Go code (compiled by Go to an object file `go.o`, but not processed any further) with your non-Go code.

It is also possibly using `cgo` from the `net` library (see https://golang.org/pkg/net/#hdr-Name_Resolution). 

Everything after `-extldflags` are flags for `gcc` (`-w` to suppress warnings IIRC).

TL;DR: I just want a static binary

_(see https://github.com/mattn/go-sqlite3/issues/384#issuecomment-290291449 for more information)_

Reply:
Agreed, thanks for the explanation.